src/Compilers/CSharp/Portable/Binder/Binder_Deconstruct.cs

```diff
@@ -488,8 +490,14 @@ private static TypeSymbol MakeMergedTupleType(ArrayBuilder<DeconstructionVariabl
                 return null;
             }
 
-            return TupleTypeSymbol.Create(locationOpt: null, elementTypes: typesBuilder.ToImmutableAndFree(), elementLocations: default(ImmutableArray<Location>),
-                                    elementNames: default(ImmutableArray<string>), compilation: compilation, diagnostics: diagnostics);
+            return TupleTypeSymbol.Create(
+                locationOpt: null,
+                elementTypes: typesBuilder.ToImmutableAndFree(),
+                elementLocations: default(ImmutableArray<Location>),
+                elementNames: default(ImmutableArray<string>),
+                compilation: compilation,
+                diagnostics: diagnostics,
+                shouldCheckConstraints: false);
```

Comment:
It looks to me that this code creates a whole new tuple type symbol, likely not identical to any tuple type symbol that was created and checked earlier. Where is that new tuple type symbol is going to be checked, including its elements that are tuple types as well?

---
In reply to: [92724771](https://github.com/dotnet/roslyn/pull/15641#discussion_r92724771) [](ancestors = 92724771,92523053)